app/org/maproulette/framework/repository/UserSavedObjectsRepository.scala

```diff
@@ -150,6 +151,41 @@ class UserSavedObjectsRepository @Inject() (
     }
   }
 
+  /**
+    * Retrieves a list of locked tasks for a specific user.
+    *
+    * @param userId       The ID of the user for whom you are requesting the saved challenges.
+    * @param limit        The maximum number of tasks to return.
+    * @param c            An optional existing connection.
+    * @return A list tasks the user has locked, each item containing the task ID, its locked time, and the challenge name.
+    */
+  def getLockedTasks(
+      userId: Long,
+      limit: Long
+  )(implicit c: Option[Connection] = None): List[LockedTaskData] = {
+    this.withMRTransaction { implicit c =>
+      val parser = for {
+        id         <- get[Long]("id")
+        parent     <- get[Long]("tasks.parent_id")
+        parentName <- get[Option[String]]("challenges.challenge_name")
+        lockedTime <- get[DateTime]("locked.locked_time")
+      } yield (LockedTaskData(id, parent, parentName, lockedTime))
+
+      val query = """
+                    SELECT t.id, t.parent_id, l.locked_time, c.name AS challenge_name
+                    FROM tasks t
+                    INNER JOIN locked l ON t.id = l.item_id
+                    LEFT JOIN challenges c ON t.parent_id = c.id
```

Comment:
task.parent_id is a FK to challenge.id, so the `LEFT` is misleading and does nothing. Replace it with `INNER`.